resip/stack/TransportSelector.cxx

```diff
@@ -1547,15 +1547,14 @@ TransportSelector::findTransportBySource(Tuple& search, const SipMessage* msg) c
 {
    DebugLog(<< "findTransportBySource(" << search << ")");
 
-   if(msg && 
-      !msg->getTlsDomain().empty() && 
-      isSecure(search.getType()))
+   if(msg && isSecure(search.getType()))
    {
       // We should not be willing to attempt sending on a TLS/DTLS/WSS transport 
       // that does not have the cert we're attempting to use, even if the 
       // IP/port/proto match. If we have not specified which identity we want
       // to use, then proceed with the code below.
-      return findTlsTransport(msg->getTlsDomain(),search.getType(),search.ipVersion());
+      return findTlsTransport(msg->getTlsDomain(), search.getType(), search.ipVersion(),
+                              search.presentationFormat(), search.getPort());
```

Comment:
I'm debating if using search.getSockaddr vs presentationFormat would be better / more efficient.  There is sample sockaddr compare code in Tuple::AnyPortCompare::operator().  Have you given this any thought?

From what I can gleam - the comparison of the presentation format string may have similar cost to doing a sockaddr compare, however the creation of the presentationFormat string bears some cost (inet_ntop), that may indicate sockaddr would be more efficient here.

Reply:
I think you are right. I am going to update it today.
To be honest, I missed this point. Thank you for the advice.